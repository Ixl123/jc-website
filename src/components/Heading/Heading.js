import React from 'react';
import { oneOf, any } from 'prop-types';

export const APPEARANCES = {
  H1: `H1`,
  H2: `H2`,
  H3: `H3`,
  H4: `H4`,
  H5: `H5`,
  H6: `H6`,
};

const Heading = ({ appearance, children, ...rest }) => {
  const { ...headingProps } = rest;
  return React.createElement(
    appearance.toLowerCase(),
    {
      ...headingProps,
    },
    children
  );
};

Heading.displayName = `Heading`;

Heading.propTypes = {
  /** any nodes to be rendered (usually text nodes) */
  children: any,

  /** typography of the heading */
  appearance: oneOf(Object.keys(APPEARANCES)),
};

Heading.defaultProps = {
  appearance: APPEARANCES.H1,
};

export default Heading;
