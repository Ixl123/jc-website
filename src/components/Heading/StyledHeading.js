import styled from 'styled-components';
import theme from 'styled-theming';

const HeadingTextShadow = theme('mode', {
  light: '2px 2px 0 rgba(0,0, 0,0.3)',
  dark: '2px 2px 0 rgba(255,255, 255,0.3)',
});

const HeadingTextColor = theme('mode', {
  light: props => props.theme.colors.black,
  dark: props => props.theme.colors.white,
});

const StyledHeadingH1 = styled.h1`
  font-size: 8rem;
  transform: skew(-5deg) rotate(-1deg);
  margin-bottom: 2rem;
  color: ${HeadingTextColor};
  text-shadow: ${HeadingTextShadow};
  a {
    color: ${HeadingTextColor};
    transition: color ${props => props.theme.themeTransition};
  }
  a:hover {
    color: ${props => props.theme.colors.primary};
    transition: none;
  }
  :before {
    width: 0;
    height: 0;
    opacity: 0.2;
    border-left: 25px solid transparent;
    border-right: 25px solid transparent;
    border-bottom: 50px solid ${props => props.theme.colors.white};
    content: '';
    pointer-events: none;
    position: absolute;
    z-index: -1;
    transform: translateX(-0.5em) translateY(-1.5rem);
  }
`;

const StyledHeadingH2 = styled(StyledHeadingH1)`
  font-size: 5rem;
`;

const StyledHeadingH3 = styled(StyledHeadingH1)`
  font-size: 4rem;
`;
const StyledHeadingH4 = styled(StyledHeadingH1)`
  font-size: 3rem;
`;
const StyledHeadingH5 = styled(StyledHeadingH1)`
  font-size: 2.5rem;
`;
const StyledHeadingH6 = styled(StyledHeadingH1)`
  font-size: 2rem;
`;
