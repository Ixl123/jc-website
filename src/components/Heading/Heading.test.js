import React from 'react';
import { render, cleanup } from 'react-testing-library';
import 'jest-dom/extend-expect';
import Heading, { APPEARANCES } from './Heading';
// automatically unmount and cleanup DOM after the test is finished.
afterEach(cleanup);

Object.keys(APPEARANCES).forEach(appearance => {
  test(`should render a ${appearance} header tags`, () => {
    const { container } = render(<Heading appearance={appearance}>I am a {appearance} headline</Heading>);
    expect(container.firstChild).toMatchSnapshot();
  });
});
