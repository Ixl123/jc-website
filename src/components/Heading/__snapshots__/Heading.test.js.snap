// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render a H1 header tags 1`] = `
<h1>
  I am a 
  H1
   headline
</h1>
`;

exports[`should render a H2 header tags 1`] = `
<h2>
  I am a 
  H2
   headline
</h2>
`;

exports[`should render a H3 header tags 1`] = `
<h3>
  I am a 
  H3
   headline
</h3>
`;

exports[`should render a H4 header tags 1`] = `
<h4>
  I am a 
  H4
   headline
</h4>
`;

exports[`should render a H5 header tags 1`] = `
<h5>
  I am a 
  H5
   headline
</h5>
`;

exports[`should render a H6 header tags 1`] = `
<h6>
  I am a 
  H6
   headline
</h6>
`;
